core/Config.php

```diff
@@ -514,7 +514,7 @@ protected function getUncached($class, $name, $sourceOptions, &$result, $suppres
 		// Include extensions only if not flagged not to, and some have been set
 		if (($sourceOptions & self::EXCLUDE_EXTRA_SOURCES) != self::EXCLUDE_EXTRA_SOURCES) {
 			// If we don't have a fresh list of extra sources, get it from the class itself
-			if (!array_key_exists($class, $this->extraConfigSources)) {
+			if (!isset($class[$this->extraConfigSources]) || $class[$this->extraConfigSources] === null) {
```

Comment:
The Optimizer plan is going to come as a seperate PR. Just wondering, since this PHP file is meant to be run from CLI on the entire framework codebase, should I have two commits with just the Optimizer::optimize() / Optimizer::deoptimize() and then the next one that touches whatever files use is_array(), array_key_exists(), etc? or just one for it all?